### PR TITLE
Feature/townhall citizen info

### LINKED ISSUE #2407: [Feature Request] TH to show Citizen's Skills
TH Citizens Tab in GUI currently shows the "Names" of the citizens.

Request: On click of citizen's name it shows the "Skills" for that specific Citizen (on the other page of the GU).
For better management of Workers.

th_gui

### LINKED ISSUE #2428: miner
5-5-2018 Minecolonies version

version X.X.XXXX or curse X.X.XX

miner Expected behavior

- miner worked like in the prefious patch

### Actual behaviour
miner removes all ladders to place them on another side then he places a lot of cobbelstone and suficates in it

- YYY

### Steps to reproduce the problem

1. XYZ
2. ZYX
3. ...
4. Profit

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2407
Closes #2428

# Changes proposed in this pull request:
- Add info GUI to townhall with recall button for individual citizen

Review please

### CODE REVIEW

**Review by marchermans — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 1]
```diff
@@ -367,8 +367,8 @@ public void onOpened()
 
         createHealthBar();
         createSaturationBar();
-        createXpBar();
-        createSkillContent();
+        createXpBar(citizen, this);
+        createSkillContent(citizen, this);
 
         resourceList.setDataProvider(new ScrollingList.DataProvider()
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:373]: MAJOR Reduce this anonymous class number of lines from 65 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)

File: src/main/java/com/minecolonies/coremod/network/messages/RecallSingleCitizenMessage.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package com.minecolonies.coremod.network.messages;
+
+import com.minecolonies.api.colony.permissions.Action;
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.LanguageHandler;
+import com.minecolonies.coremod.colony.CitizenData;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/network/messages/RecallSingleCitizenMessage.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.buildings.AbstractBuilding'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/network/messages/RecallSingleCitizenMessage.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/network/messages/RecallSingleCitizenMessage.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/network/messages/RecallSingleCitizenMessage.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import com.minecolonies.coremod.colony.buildings.BuildingTownHall;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/network/messages/RecallSingleCitizenMessage.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.buildings.BuildingTownHall'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 66 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L364](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L364): Refactor this method to reduce its Cognitive Complexity from 23 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L395](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L395): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L396](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L396): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L602](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L602): Split this 183 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L48](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L48): class "WindowTownHall" has 40 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L864](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L864): Reduce this anonymous class number of lines from 41 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/18fc97f512fa819224dc2e9eb7096909881fd9c4/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 63 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L364](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L364): Refactor this method to reduce its Cognitive Complexity from 23 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L395](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L395): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L396](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L396): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L602](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L602): Split this 183 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L48](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L48): class "WindowTownHall" has 40 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L864](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L864): Reduce this anonymous class number of lines from 41 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/5cc7eaae46918f63ac1b57ddcb2671c1bf3f05e9/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**